File: androld/app/src/main/java/z3d/base/ObjData.java
package z3d.base;


import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.nio.FloatBuffer;
import java.nio.ShortBuffer;
import java.util.ArrayList;
import java.util.List;

import z3d.engine.ResCount;




public class ObjData extends ResCount {

    public List<Float> verticeslist;
    public List<Float> uvlist;
    public List<Short> indexs;
    public List<Float> lightuvs;
    public List<Float> normals;
    public List<Float> tangents;
    public List<Float> bitangents;


    /**顶点 uv lightuv normal 合成一个 va */
    public boolean compressBuffer;
    public int uvsOffsets;
    public int lightuvsOffsets;
    public int normalsOffsets;
    public int tangentsOffsets;
    public int bitangentsOffsets;
    public int stride;
    public boolean hasdispose;
    public boolean isCompile;


    public FloatBuffer vertexBuffer;
    public FloatBuffer uvBuffer;
    public ShortBuffer indexBuffer;
    public FloatBuffer  lightUvBuffer;
    public FloatBuffer  normalsBuffer;
    public FloatBuffer  tangentBuffer;
    public FloatBuffer  bitangentBuffer;
    public int treNum ;


    public void makeTriModel(){


        this.verticeslist=new ArrayList<Float>();//结果顶点坐标列表
        this.verticeslist.add(-1f);
        this.verticeslist.add(0f);
        this.verticeslist.add(0f);

        this.verticeslist.add(3f);
        this.verticeslist.add(0f);
        this.verticeslist.add(0f);

        this.verticeslist.add(3f);
        this.verticeslist.add(5f);
        this.verticeslist.add(0f);

        this.verticeslist.add(-1f);
        this.verticeslist.add(4f);
        this.verticeslist.add(0f);

        this.verticeslist.add(-1f);
        this.verticeslist.add(0f);
        this.verticeslist.add(0f);


        this.indexs=new ArrayList<Short>();//结果顶点坐标列表
        this.indexs.add((short)0);
        this.indexs.add((short)1);
        this.indexs.add((short)2);



        this.upToGup();

    }
    public void  upToGup()
    {
        if(!this.isCompile){
            this.vertexBuffer=this.upGpuvertexBufferbbb(this.verticeslist);
            if(this.normals!=null&&this.normals.size()>0){
                this.normalsBuffer=this.upGpuvertexBufferbbb(this.normals);
            }
            this.indexBuffer=this.upGpuIndexBuffercopy(this.indexs);
            this.treNum= this.indexs.size();
            this.isCompile=true;
        }


    }
/*
short Indexbuff
 */
    public  ShortBuffer upGpuIndexBuffercopy(List<Short> data){
        int size=data.size();
        ByteBuffer buffer=ByteBuffer.allocateDirect(size*2);
        buffer.order(ByteOrder.nativeOrder());
        ShortBuffer verBuff=buffer.asShortBuffer();
        for (int i=0;i<size;i++){
            verBuff.put(data.get(i));
        }
        verBuff.position(0);
        return verBuff;

    }
    /*
    float vertexBuff
     */
    public  FloatBuffer upGpuvertexBufferbbb(List<Float> data){
        int size=data.size();
        ByteBuffer buffer=ByteBuffer.allocateDirect(size*4);
        buffer.order(ByteOrder.nativeOrder());
        FloatBuffer verBuff=buffer.asFloatBuffer();
        for (int i=0;i<size;i++){
            verBuff.put(data.get(i));
        }
        verBuff.position(0);
        return verBuff;

    }



}

File: h5gl/new/src/pan3d/base/ObjData.ts
module Pan3d {
    export class ObjData extends ResCount {
        public vertices: Array<number> = new Array;
        public uvs: Array<number> = new Array;
        public indexs: Array<number> = new Array;
        public lightuvs: Array<number> = new Array;
        public normals: Array<number> = new Array;
        public tangents: Array<number> = new Array;
        public bitangents: Array<number> = new Array;


        public treNum: number;

        public vertexBuffer: WebGLBuffer;
        public uvBuffer: WebGLBuffer;
        public indexBuffer: WebGLBuffer;
        public lightUvBuffer: WebGLBuffer;
        public normalsBuffer: WebGLBuffer;
        public tangentBuffer: WebGLBuffer;
        public bitangentBuffer: WebGLBuffer;


        constructor(value: Scene3D) {
            super(value);
        }
        public upToGpu(): void {


            if (this.indexs.length) {
                this.treNum = this.indexs.length
                var context3D: Context3D = this.scene3D.context3D;
                this.vertices ? this.vertexBuffer = context3D.uploadBuff3D(this.vertices) : null;
                this.uvs ? this.uvBuffer = context3D.uploadBuff3D(this.uvs) : null;
                this.normals ? this.normalsBuffer = context3D.uploadBuff3D(this.normals) : null;
                this.indexs ? this.indexBuffer = context3D.uploadIndexBuff3D(this.indexs) : null;
            }

        }



    }
}